lib/MusicBrainz/Server/Edit/Medium/Edit.pm

```diff
@@ -381,17 +385,17 @@ sub accept {
                 [ name => \@merged_names ],
                 [ recording_id => \@merged_recordings ],
                 [ length => \@merged_lengths ],
-                [ artist_credit => \@merged_artist_credits, \&hash_artist_credit ],
+                [ artist_credit => \@merged_artist_credits, $hash_artist_credit ],
                 [ is_data_track => \@merged_is_data_tracks ]
             ) {
                 my ($property, $container, $key_generation) = @$merge;
-                push @$container, merge(
-                    track_column($property, $self->data->{old}{tracklist}),
-                    track_column($property, $current_tracklist),
-                    track_column($property, $data_new_tracklist),
-                    { CONFLICT => sub { die } },
-                    $key_generation // ()
```

Comment:
`Text::Diff3::merge` doesn't appear to have key support, which is why we're now feeding it through `track_column` to call it in there, and mapping the artist credit keys to the actual AC instances above in `%hashed_artist_credits`.